Swift/Collection.swift

```diff
@@ -58,21 +58,21 @@ public final class Collection : CollectionChangeObservable, Indexable {
     public static let defaultCollectionName: String = "_default"
     
     /// Collection name.
-    var name: String { _impl.name }
+    public var name: String { _impl.name }
     
     /// The scope of the collection.
-    var scope: Scope { Scope(_impl.scope) }
+    public var scope: Scope { Scope(_impl.scope) }
```

Comment:
You can keep the scope inside the Collection so that it will not create a new Scope every time.